astropy/units/tests/test_quantity.py

```diff
@@ -138,6 +138,28 @@ def test_preserve_dtype(self):
         q5 = u.Quantity(decimal.Decimal('10.25'), u.m, dtype=object)
         assert q5.dtype == object
 
+    def test_numpy_style_dtype_inspect(self):
+        """Test that if ``dtype=None``, NumPy's dtype inspection is used."""
+        q2 = u.Quantity(12, dtype=None)
+        assert np.issubdtype(q2.dtype, np.integer)
+
+    def test_float_dtype_promotion(self):
+        """Test that if ``dtype=numpy.floating``, the minimum precision is float64."""
+        q1 = u.Quantity(12, dtype=np.floating)
+        assert not np.issubdtype(q1.dtype, np.integer)
+        assert q1.dtype == np.float64
+
+        q2 = u.Quantity(np.float64(12), dtype=np.floating)
+        assert q2.dtype == np.float64
+
+        if hasattr(np, "float16"):
```

Comment:
Maybe explicitly test `float32` (which always is present) too?

Reply:
✅